modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v14_0_0/ObjectDefinitionUpgradeProcess.java

```diff
@@ -0,0 +1,172 @@
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectFieldLocalService;
+import com.liferay.object.service.ObjectRelationshipLocalService;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Pedro Leite
+ */
+public class ObjectDefinitionUpgradeProcess extends UpgradeProcess {
+
+	public ObjectDefinitionUpgradeProcess(
+		CompanyLocalService companyLocalService,
+		ObjectDefinitionLocalService objectDefinitionLocalService,
+		ObjectFieldLocalService objectFieldLocalService,
+		ObjectRelationshipLocalService objectRelationshipLocalService) {
+
+		_companyLocalService = companyLocalService;
+		_objectDefinitionLocalService = objectDefinitionLocalService;
+		_objectFieldLocalService = objectFieldLocalService;
+		_objectRelationshipLocalService = objectRelationshipLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_companyLocalService.forEachCompanyId(
+			companyId -> {
+				ObjectDefinition objectDefinition =
+					_objectDefinitionLocalService.
+						getObjectDefinitionByExternalReferenceCode(
+							"L_COMMERCE_PRODUCT_DEFINITION", companyId);
+
+				objectDefinition.setPKObjectFieldDBColumnName(
+					_NEW_PK_OBJECT_FIELD_NAME);
+				objectDefinition.setPKObjectFieldName(
+					_NEW_PK_OBJECT_FIELD_NAME);
+
+				objectDefinition =
+					_objectDefinitionLocalService.updateObjectDefinition(
+						objectDefinition);
+
+				_updateObjectField(
+					_NEW_PK_OBJECT_FIELD_NAME,
+					_objectFieldLocalService.getObjectField(
+						objectDefinition.getObjectDefinitionId(), "id"));
+				_updateObjectField(
+					_NEW_PK_OBJECT_FIELD_NAME,
+					_objectFieldLocalService.getObjectField(
+						objectDefinition.getObjectDefinitionId(), "productId"));
+
+				_updateDynamicObjectDefinitionTable(
+					objectDefinition.getExtensionDBTableName(), "LONG not null",
+					objectDefinition, _OLD_PK_OBJECT_FIELD_NAME);
+				_updateDynamicObjectDefinitionTable(
+					objectDefinition.getLocalizationDBTableName(),
+					"LONG not null", objectDefinition,
+					_OLD_PK_OBJECT_FIELD_NAME);
+
+				List<ObjectRelationship> objectRelationships =
+					_objectRelationshipLocalService.getObjectRelationships(
+						objectDefinition.getObjectDefinitionId());
+
+				for (ObjectRelationship objectRelationship :
+						objectRelationships) {
+
+					String dbTableName = null;
+					String dbType = "LONG";
+					String oldDBColumnName = null;
+
+					if (Objects.equals(
+							objectRelationship.getType(),
+							ObjectRelationshipConstants.TYPE_ONE_TO_ONE) ||
+						Objects.equals(
+							objectRelationship.getType(),
+							ObjectRelationshipConstants.TYPE_ONE_TO_MANY)) {
```

Comment:
@pedro-oliveira446 why don't we need the `dbType += " not null";` for this case?

Reply:
We only need to do this for primary-key columns. This column is part of a one-to-many relationship, so it is not a primary key.